beets/ui/__init__.py

```diff
@@ -1093,43 +1103,158 @@ def _open_library(config):
     return lib
 
 
-def _raw_main(args, lib=None):
+class BeetsCLI(click.MultiCommand):
+
+    def _commands(self, ctx):
+        # FIXME: Awful performance, cache per ctx
```

Comment:
Is this why a global context is wanted?

(PS: I didn't checkout and read, just browsing via Github...)